requester/requester.go

```diff
@@ -222,7 +232,15 @@ func (b *Work) runWorker(n int) {
 	} else {
 		tr.TLSNextProto = make(map[string]func(string, *tls.Conn) http.RoundTripper)
 	}
-	client := &http.Client{Transport: tr, Timeout: time.Duration(b.Timeout) * time.Second}
+
+	var checkRedirect func(req *http.Request, via []*http.Request) error
+	if b.DisableRedirects {
+		checkRedirect = func(req *http.Request, via []*http.Request) error {
+			return errRedirectNotAllowed
+		}
+	}
+
+	client := &http.Client{Transport: tr, Timeout: time.Duration(b.Timeout) * time.Second, CheckRedirect: checkRedirect}
```

Comment:
keep this line as it is.

then add 

if b.DisableRedirects {
   client.CheckRedirect = func (req *http.Request, via []*http.Request) error { ... } 
}